mpfs: Disable external interrupts on all harts

jlaitine opened the pull request:
Disable external interrupts on all harts

This solves unexpected isr assertion in SMP mode, if console is disabled later after boot

pussuw commented on arch/risc-v/src/mpfs/mpfs_serial.c:
```diff
@@ -1429,6 +1429,16 @@ static void up_txint(struct uart_dev_s *dev, bool enable)
   flags = enter_critical_section();
   if (enable)
     {
+      /* If the console has been disabled after boot, don't allow enabling
```
Fairly certain this is not upstreamable, enabling tx interrupts and console have nothing to do with eachother

pussuw commented on arch/risc-v/src/mpfs/mpfs_plic.h:
```diff
@@ -58,7 +58,7 @@ void mpfs_plic_init_hart(uintptr_t hartid);
  *
  ****************************************************************************/
 
-uintptr_t mpfs_plic_get_iebase(void);
```
Please keep the incoming parameter as uintptr_t 

I see that the conversion macros have wrong types (they use int for hartid), but let's not spread that rot here as well.

Hartid should be uintptr_t as its width depends on the architecture (32-bits on RV32 and 64-bits on RV64)

jlaitine commented:
Changed riscv_mhartid -> up_cpu_index to match the patch offered to upstream, added also imx9 uart fix to the same pr, as this is related to same uart disabling issue, but on nxp

jpaali reviewed (approved):
LGTM
